bodo/pandas/plan_optimizer.pyx

```diff
@@ -714,10 +714,10 @@ cdef class LogicalGetParquetRead(LogicalOperator):
 
     def __cinit__(self, object out_schema, object parquet_path, object storage_options):
         self.out_schema = out_schema
-        cdef unique_ptr[CLogicalGet] c_logical_get = make_parquet_get_node(parquet_path, out_schema, storage_options)
-        self.c_logical_operator = unique_ptr[CLogicalOperator](<CLogicalGet*> c_logical_get.release())
         self.path = parquet_path
         self.storage_options = storage_options
+        cdef unique_ptr[CLogicalGet] c_logical_get = make_parquet_get_node(parquet_path, out_schema, storage_options, self.getCardinality())
```

Comment:
Getting the exact number of rows during planning is probably too much overhead. Some approximation is probably enough. What do we do in BodoSQL @knassre-bodo ?